app/managers/websocket_manager.ts

```diff
@@ -212,48 +215,43 @@ class WebsocketManager {
     }
 
     private stopPeriodicStatusUpdates(serverUrl: string) {
-        const currentId = this.statusUpdatesIntervalIDs[serverUrl];
-        if (currentId != null) {
-            clearInterval(currentId);
-        }
-
+        clearInterval(this.statusUpdatesIntervalIDs[serverUrl]);
         delete this.statusUpdatesIntervalIDs[serverUrl];
     }
 
     private onAppStateChange = async (appState: AppStateStatus) => {
-        const isActive = appState === 'active';
-        if (isActive === this.previousActiveState) {
+        const isMain = isMainActivity();
+        if (!isMain) {
             return;
         }
 
-        const isMain = isMainActivity();
-
-        this.cancelAllConnections();
-        if (!isActive && !this.isBackgroundTimerRunning) {
-            this.isBackgroundTimerRunning = true;
-            this.cancelAllConnections();
-            this.backgroundIntervalId = BackgroundTimer.setInterval(() => {
-                this.closeAll();
-                BackgroundTimer.clearInterval(this.backgroundIntervalId!);
-                this.isBackgroundTimerRunning = false;
-            }, WAIT_TO_CLOSE);
-
-            this.previousActiveState = isActive;
+        const isActive = appState === 'active';
+        if (isActive === this.previousActiveState) {
```

Comment:
On Android, the share extension is another activity and uses RN that is why we check as any activity running with RN will emit this event and we do not really need WS for the share extension unless the app is opened.

As for the app state, it triggers on 3 events: active when the app is in foreground and the other two are background and when you place the app in the task manager (app switcher) as inactive, this code sort of follows best practices that can be found in the example in the doc https://reactnative.dev/docs/appstate

Reply:
Thanks for the explanation. I take it that the share extension has a `WebSocketManager` then even though it doesn't need the WS connection like you said

And I missed that their example also has the previous value. That still seems like it shouldn't be necessary to me because the app should never go from active to active, but I guess it's good to be safe